### PR TITLE
OSDOCS-3419: healthCheckInterval Parameter documentation

### PR DESCRIPTION
**Pull request by cowgirljessi**
Version(s):
4.11 (will need confirmation)

Issue:
https://issues.redhat.com/browse/OSDOCS-3419

Link to docs preview: https://deploy-preview-44848--osdocs.netlify.app/openshift-enterprise/latest/networking/ingress-operator.html#nw-ingress-controller-configuration-parameters_configuring-ingress

Additional information:

### CODE REVIEW

File: modules/nw-ingress-controller-configuration-parameters.adoc [hunk 1]
```diff
@@ -200,24 +200,26 @@ For request headers, these adjustments are applied only for routes that have the
...
 
-* `threadCount` specifies the number of threads to create per HAProxy process. Creating more threads allows each Ingress Controller pod to handle more connections, at the cost of more system resources being used. HAProxy
-supports up to `64` threads. If this field is empty, the Ingress Controller uses the default value of `4` threads. The default value can change in future releases. Setting this field is not recommended because increasing the number of HAProxy threads allows Ingress Controller pods to use more CPU time under load, and prevent other pods from receiving the CPU resources they need to perform. Reducing the number of threads can cause the Ingress Controller to perform poorly.
+* `healthCheckInterval` defines how long the router waits between health checks. The default is `30s`.
```
Comment (candita) [inline modules/nw-ingress-controller-configuration-parameters.adoc:211]:
```suggestion
* `healthCheckInterval` defines how long the router waits between health checks. The default is `5s`.
```

File: modules/nw-ingress-controller-configuration-parameters.adoc [hunk 2]
```diff
@@ -200,24 +200,26 @@ For request headers, these adjustments are applied only for routes that have the
...
 
-* `clientTimeout` specifies how long a connection is held open while waiting for a client response. If unset, the default timeout is `30s`.
+* `serverFinTimeout` specifies how long a connection is held open while waiting for the server response to the client that is closing the connection. The default timeout is `1s`.
 
-* `serverFinTimeout` specifies how long a connection is held open while waiting for the server response to the client that is closing the connection. If unset, the default timeout is `1s`.
+* `serverTimeout` specifies how long a connection is held open while waiting for a server response. The default timeout is `30s`.
 
-* `serverTimeout` specifies how long a connection is held open while waiting for a server response. If unset, the default timeout is `30s`.
+* `tlsInspectDelay` specifies how long the router can hold data to find a matching route. Setting this value too short can cause the router to fall back to the default certificate for edge-terminated, reencrypted, or passthrough routes, even when using a better matched certificate. The default inspect delay is `5s`.
```
Comment (candita) [inline modules/nw-ingress-controller-configuration-parameters.adoc:217]: tlsInspectDelay is alphabetically after threadCount

**Comment by cowgirljessi** [comment]
@candita I resolved your comments and will move on to QA/Peer review processes; other comments welcome as needed

**Comment by cowgirljessi** [comment]
@ShudiLi can I get a QA check please?

**Comment by ShudiLi** [comment]
/label qe-approved
@cowgirljessi thanks

File: modules/nw-ingress-controller-config-tuningoptions-healthcheckinterval [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * ingress/configure-ingress-operator.adoc
```
Comment (skrthomas) [inline modules/nw-ingress-controller-config-tuningoptions-healthcheckinterval:3]: this is just a commented section, but I this needs to be `networking/ingress-operator.adoc`

File: modules/nw-ingress-controller-configuration-parameters.adoc [hunk 4]
```diff
@@ -197,24 +197,26 @@ For request headers, these adjustments are applied only for routes that have the
...
 
-* `threadCount` specifies the number of threads to create per HAProxy process. Creating more threads allows each Ingress Controller pod to handle more connections, at the cost of more system resources being used. HAProxy
-supports up to `64` threads. If this field is empty, the Ingress Controller uses the default value of `4` threads. The default value can change in future releases. Setting this field is not recommended because increasing the number of HAProxy threads allows Ingress Controller pods to use more CPU time under load, and prevent other pods from receiving the CPU resources they need to perform. Reducing the number of threads can cause the Ingress Controller to perform poorly.
+* `healthCheckInterval` defines how long the router waits between health checks. The default is `5s`.
```
Comment (jeana-redhat) [inline modules/nw-ingress-controller-configuration-parameters.adoc:208]: Just for consistency's sake :slightly_smiling_face: 
```suggestion
* `healthCheckInterval` specifies how long the router waits between health checks. The default interval is `5s`.
```

**Review by jeana-redhat — COMMENTED** [review]
A few minor things, otherwise looks good

File: modules/nw-ingress-controller-config-tuningoptions-healthcheckinterval.adoc [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * networking/ingress-operator.adoc
+
+:_content-type: PROCEDURE
+[id="nw-ingress-controller-config-tuningoptions-healthcheckinterval_{context}"]
+= Setting Ingress Controller Health Check Interval
```
[the remark this answers is not in the document]
Comment (cowgirljessi) [reply modules/nw-ingress-controller-config-tuningoptions-healthcheckinterval.adoc:7]: Changed!

File: modules/nw-ingress-controller-config-tuningoptions-healthcheckinterval.adoc [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+* Update the Ingress Controller to change the interval between back end health checks:
++
+[source,terminal]
+----
+$ oc -n openshift-ingress-operator patch ingresscontroller/default --type=merge -p '{"spec":{"tuningOptions": {"healthCheckInterval": 8}}}'
+----
++
+[NOTE]
+====
+If you want to override the `healthCheckInterval` for a single route, it can be done for an individual route by using route annotation "router.openshift.io/haproxy.health.check.interval"
```
[the remark this answers is not in the document]
Comment (cowgirljessi) [reply modules/nw-ingress-controller-config-tuningoptions-healthcheckinterval.adoc:23]: Replaced with backticks; thank you for the minimalistic approach.

File: modules/nw-ingress-controller-configuration-parameters.adoc (hunk 4 above)
Comment (cowgirljessi) [reply modules/nw-ingress-controller-configuration-parameters.adoc]: Indeed

File: modules/nw-ingress-controller-config-tuningoptions-healthcheckinterval.adoc [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+.Prerequisites
+* The following assumes that you already created an Ingress Controller.
+
+.Procedure
+* Update the Ingress Controller to change the interval between back end health checks:
++
+[source,terminal]
+----
+$ oc -n openshift-ingress-operator patch ingresscontroller/default --type=merge -p '{"spec":{"tuningOptions": {"healthCheckInterval": 8}}}'
```
[the remark this answers is not in the document]
Comment (candita) [reply modules/nw-ingress-controller-config-tuningoptions-healthcheckinterval.adoc:18]: Yes, it is  required to add a unit afterward.

**Comment by ahardin-rh** [comment]
@cowgirljessi Is this ready to merge? Looks like https://github.com/openshift/cluster-ingress-operator/pull/712 is merged. Thanks!

**Comment by cowgirljessi** [comment]
@ahardin-rh just have one outstanding question and then yes

File: modules/nw-ingress-controller-config-tuningoptions-healthcheckinterval.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (cowgirljessi) [reply modules/nw-ingress-controller-config-tuningoptions-healthcheckinterval.adoc:18]: Updated

**Comment by cowgirljessi** [comment]
@ShudiLi one more review pretty please

**Comment by ShudiLi** [comment]
@cowgirljessi @ahardin-rh  please make sure, 8s is replaced with "8s" in line 18:
$ oc -n openshift-ingress-operator patch ingresscontroller/default --type=merge -p '{"spec":{"tuningOptions": {"healthCheckInterval": 8s}}}'

It should be:
$ oc -n openshift-ingress-operator patch ingresscontroller/default --type=merge -p '{"spec":{"tuningOptions": {"healthCheckInterval": "8s"}}}'

**Comment by ahardin-rh** [comment]
@ShudiLi Updated. Thank you!

**Pull request merged**